Test/Load/setup-files/LoadTestCartData.php

```diff
@@ -0,0 +1,217 @@
+ */
+class LoadTestCartData implements LoadTestCartDataInterface
+{
+    const E_BOLT_GENERAL_ERROR = 2001001;
+
+    /**
+     * @var HookHelper
+     */
+    private $hookHelper;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var Request
+     */
+    private $request;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /*
+     * @var CheckoutSession
+     */
+    private $checkoutSession;
+
+    /*
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /*
+     * @var Product
+     */
+    private $product;
+
+    /*
+     * @var QuoteFactory
+     */
+    private $quoteFactory;
+
+    /*
+     * @var CartItemInterfaceFactory
+     */
+    private $cartItemFactory;
+
+    /**
+     * @param HookHelper                $hookHelper
+     * @param CartHelper                $cartHelper
+     * @param LogHelper                 $logHelper
+     * @param Request                   $request
+     * @param Response                  $response
+     * @param ProductRepository         $productRepository
+     * @param CheckoutSession           $checkoutSession
+     * @param CartItemInterfaceFactory  $cartItemFactory
+     * @param Product                   $product
+     * @param QuoteFactory              $quoteFactory
+     */
+    public function __construct(
+        HookHelper $hookHelper,
+        CartHelper $cartHelper,
+        LogHelper $logHelper,
+        Request $request,
+        Response $response,
+        ProductRepository $productRepository,
+        CheckoutSession $checkoutSession,
+        CartItemInterfaceFactory $cartItemFactory,
+        Product $product,
+        QuoteFactory $quoteFactory
+    )
+    {
+        $this->product = $product;
+        $this->quoteFactory = $quoteFactory;
+        $this->checkoutSession = $checkoutSession;
+        $this->hookHelper = $hookHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->response = $response;
+        $this->cartHelper = $cartHelper;
+        $this->productRepository = $productRepository;
+        $this->cartItemFactory = $cartItemFactory;
+    }
+
+    /**
+     * Load-test hook: Populate cart and create bolt order.
+     *
+     * @api
+     * @inheritDoc
+     * @see CartDataInterface
+     */
+    public function execute($cart = null) {
+        try {
+            $payload = $this->request->getContent();
+
+            $this->logHelper->addInfoLog('[-= LoadTest endpoint =-]');
+            $this->logHelper->addInfoLog($payload);
+
+            if (empty($cart)) {
+                throw new BoltException(
+                    __('Missing cart data.'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $this->preProcessWebhook();
+
+            // Create cart with the given products
```

Comment:
For my own understanding here, you say you're creating a cart with given products, but it looks like you're creating a quote from the cart variable passed in. Is this just naming convention in m2?

Reply:
Yeah so I believe in M2 a quote is similar to a cart. And so we just add the items to the quote.